util/buffer.go

```diff
@@ -0,0 +1,40 @@
+package util
+
+import (
+	"bytes"
+	"sync"
+)
+
+// NewLocalBuffer returns a thread-safe Read/Write closer.
+func NewLocalBuffer(b bytes.Buffer) *LocalBuffer { return &LocalBuffer{Buffer: b} }
+
+// LocalBuffer provides a thread-safe in-memory buffer.
+type LocalBuffer struct {
+	Buffer bytes.Buffer
```

Comment:
Should this be unexported?

Reply:
It doesn't really matter - I changed it to be unexported.